microsoft/testsuites/vm_extensions/cvt.py

```diff
@@ -0,0 +1,284 @@
+
+from lisa import (
+    CustomScript,
+    CustomScriptBuilder,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    create_timer,
+    schema,
+    simple_requirement,
+)
+from lisa.executable import ExecutableResult
+from lisa.features import Disk
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.tools import Find, Mkdir, Wget
+from lisa.util import SkippedException
+
+
+def _init_disk(log: Logger, node: Node) -> None:
+    disk = node.features[Disk]
+    log.info("Adding 1st managed disk of size 1GB")
+    disk.add_data_disk(1, schema.DiskType.PremiumSSDLRS, 1)
+    log.info("Adding 2nd managed disk of size 10GB")
+    disk.add_data_disk(1, schema.DiskType.PremiumSSDLRS, 10)
+
+
+def _get_extension_name(log: Logger, os: str) -> str:
+    # UBUNTU-22.04-64
+    distro = os[:-3]
+    distro = "".join(re.findall(r"[A-Z0-9]", distro))
+    extension_name = "Linux" + distro
+    log.info(f"Extension name : '{extension_name}'")
+    return extension_name
+
+
+def _get_os_info_from_extension(log: Logger, node: Node) -> str:
+    task_id = str(uuid.uuid4())
+    cur_time = datetime.datetime.now().isoformat() + "Z"
+    extension_name = "Microsoft.Azure.RecoveryServices.SiteRecovery.Linux"
+    settings = {
+        "publicObject": "",
+        "module": "a2a",
+        "timeStamp": cur_time,
+        "commandToExecute": "GetOsDetails",
+        "taskId": task_id,
+    }
+    extension = node.features[AzureExtension]
+
+    result = extension.create_or_update(
+        name=extension_name,
+        publisher="Microsoft.Azure.RecoveryServices.SiteRecovery",
+        type_="Linux",
+        type_handler_version="1.0",
+        auto_upgrade_minor_version=True,
+        settings=settings,
+    )
+    assert_that(result["provisioning_state"]).described_as(
+        "Expected the extension to succeed"
+    ).is_equal_to("Succeeded")
+
+    extension_instance_view = extension.get(
+        name=extension_name,
+    ).instance_view
+
+    extension_substatus = ""
+    for substatus in extension_instance_view.substatuses:
+        log.info(f"Substatus : '{substatus}'")
+        code = substatus.code
+        if code is not None and code.find("platform") != -1:
+            extension_substatus = code
+            break
+    if not extension_substatus:
+        return extension_substatus
+
+    log.info(f"Substatus : '{extension_substatus}'")
+    split_status = re.split(",|/", extension_substatus)
+    for component_status in split_status:
+        if component_status is not None:
+            if component_status.find("osidentifier") != -1:
+                log.info(f"Component status : '{component_status}'")
+                os_encoded = re.split(":", component_status)[-1]
+                os = base64.b64decode(os_encoded).decode()
+
+    log.info(f"os : '{os}'")
+    return os
+
+
+def _install_asr_extension_distro(log: Logger, node: Node, os: str) -> None:
+    task_id = str(uuid.uuid4())
```

Comment:
Can you refer here https://github.com/microsoft/lisa/blob/main/microsoft/testsuites/vm_extensions/azureperformancediagnostics.py#L48, skip cases if the distro is not supported by the extension in before_case, also get `cvtbinaries_sasuri` should also happen in before_case